apps/server/src/shared/domain/rules/task.rule.spec.ts

```diff
@@ -101,8 +102,9 @@ describe('TaskRule', () => {
 
 	describe('User [STUDENT]', () => {
 		let student: User;
+		let role: Role;
```

Comment:
Can you please update the existing code to also follow the newer test setups?
describe with when
remove beforeEach and add setup function
and so on

Reply:
There is a secondary PR waiting on top of this https://github.com/hpi-schul-cloud/schulcloud-server/pull/4071 
I would like to move updating the test structure there.
Also refactoring all task test should be done in a separate PR as this ticket is currently blocking other work from been completed.